api/src/pcapi/core/offers/repository.py

```diff
@@ -182,23 +182,43 @@ def get_capped_offers_for_filters(
     return offers
 
 
+def get_offers_by_date_field_range(
+    date_field: str, lower_bound: datetime.datetime, upper_bound: datetime.datetime
+) -> sa_orm.Query:
+    column = getattr(models.Offer, date_field)
+    return db.session.query(models.Offer).filter(
+        column != None,
+        column >= lower_bound,
+        column <= upper_bound,
+    )
+
+
 def get_offers_by_publication_date(
```

Comment:
```suggestion
def get_offers_by_publication_datetime(
```
On peut aussi en profiter pour renommer la fonction :)